src/c3_fmt.c

```diff
@@ -350,7 +350,8 @@ static int valid(char *ciphertext, struct fmt_main *self)
 static void *binary(char *ciphertext)
 {
 	static char out[BINARY_SIZE];
-	strncpy(out, ciphertext, sizeof(out)); /* NUL padding is required */
+	strncpy(out, ciphertext, sizeof(out)-1);
+	out[sizeof(out)-1] = 0;
```

Comment:
Please at least restore the comment. The original code was also more correct, but I understand that gcc might be correctly thinking that `strncpy()`, given its name, is usually misused for producing NUL-terminated strings...

Reply:
No, the comment means that strncpy's NUL padding property is actually required here. The rest of this format depends on NUL padding (rather than NUL termination) of this buffer. I included this comment because most code out there (not in JtR core, but in the world at large) misuses `strncpy` because it's a misnamed function. So when we use it properly, I mention that, to ensure that no one would change this use of it to e.g. our `strnzcpy` later.